src/couch/src/couch_os_process.erl

```diff
@@ -256,6 +256,24 @@ code_change(_, {os_proc, Cmd, Port, W, R, Timeout} , _) ->
 code_change(_OldVsn, State, _Extra) ->
     {ok, State}.
 
+format_status(_Opt, [_PDict, #os_proc{} = State]) ->
+    % command field can potentially contain secrets
+    #os_proc{
+        port = Port,
+        writer = Writer,
+        reader = Reader,
+        timeout = TimeOut,
+        idle = Idle
+    } = State,
+    MapState = #{
+        port => couch_term:format_port(Port),
+        writer => couch_term:format_fun(Writer),
+        reader => couch_term:format_fun(Reader),
+        timeout => TimeOut,
+        idle => Idle
+    },
+    [{data, [{"State", MapState}]}].
```

Comment:
it is odd that we convert everything to maps for format_status even though that is not the format it takes really.